network/wsPeer.go

```diff
@@ -391,6 +392,32 @@ func (wp *wsPeer) IPAddr() []byte {
 	return result
 }
 
+// RoutingAddr returns meaningful routing part of the address:
+// ipv4 for ipv4 addresses
+// top 8 bytes of ipv6 for ipv6 addresses
+// low 4 bytes for ipv4 embedded into ipv6
+// see http://www.tcpipguide.com/free/t_IPv6IPv4AddressEmbedding.htm for details.
+func (wp *wsPeer) RoutingAddr() []byte {
+	isZeros := func(ip []byte) bool {
+		for i := 0; i < len(ip); i++ {
+			if ip[i] != 0 {
+				return false
+			}
+		}
+		return true
+	}
+
+	ip := wp.IPAddr()
+	if len(ip) == 0 || len(ip) == net.IPv4len || len(ip) != net.IPv6len {
+		return ip
+	}
```

Comment:
`len(ip) != net.IPV6len` takes care of the other two predicates

Reply:
done